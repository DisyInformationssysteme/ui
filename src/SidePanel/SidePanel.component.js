import React, { PropTypes } from 'react';
import { SidePanel as PureSidePanel } from 'react-talend-components';
import { api } from 'react-cmf';

/**
 * Checkout the {@link http://talend.github.io/react-talend-containers/examples/build/#/SidePanel|examples}
 * @param {object} props react props
 */
class SidePanel extends React.Component {
	constructor(props) {
		super(props);
		this.state = { docked: false };
	}

	render() {
		const actions = this.props.actions || [];
		const resolvedActions = actions.map((action) => {
			const info = api.action.getActionInfo(this.context, action);
			return {
				label: info.name,
				icon: info.icon,
				onClick: () => {
					this.context.store.dispatch(info);
				},
			};
		});
		const newProps = {
			actions: resolvedActions,
			docked: this.state.docked,
			onToggleDock: () => {
				this.setState({ docked: !this.state.docked });
			},
		};
		return (<PureSidePanel {...newProps} />);
	}
}

SidePanel.propTypes = {
	actions: PropTypes.arrayOf(
		PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.object,
		])
	).isRequired,
};
SidePanel.contextTypes = {
	store: React.PropTypes.object,
};

export default SidePanel;
