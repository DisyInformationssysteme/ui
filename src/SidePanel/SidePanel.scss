// SidePanel styles
$tlnd-side-menu-bg-color: #607D8B !default;
$tlnd-side-menu-txt-color: #fff !default;
$tlnd-side-menu-toggle-color: #ddd !default;

.sidePanel{
  background-color: $tlnd-side-menu-bg-color;
  height: 100%;
  .sidePanelLink, .sidePanelLink:hover{
    color: $tlnd-side-menu-txt-color;
    padding: 20px;
    text-align: left;
    display: block;
    font-size: 1.3em;
    > span {
      text-transform: uppercase;
      margin-left: 10px;
    }
  }
  .toggleBtn{
    display: block;
    color: $tlnd-side-menu-toggle-color;
    padding: 20px;
  }
}
