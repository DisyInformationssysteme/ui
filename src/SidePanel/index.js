import { themr } from 'react-css-themr';
import { SIDE_PANEL } from '../identifiers';
import SidePanel from './SidePanel.component';
import theme from './SidePanel.scss';

const ThemedSidePanel = themr(SIDE_PANEL, theme)(SidePanel);

export default ThemedSidePanel;
