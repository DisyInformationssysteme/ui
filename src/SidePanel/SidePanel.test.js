import React from 'react';
import renderer from 'react-test-renderer';
import stubContext from 'react-stub-context';
import SidePanel from './SidePanel.component';

jest.mock('react-dom');

describe('SidePanel', () => {
	it('should render provided actions', () => {
		const actions = [{
			id: 'test',
			name: 'test',
			type: 'TEST_ACTION',
		}];
		const TestSidePanel = stubContext(SidePanel, {
			router: {},
			store: {
				getState() {
					return { cmf: {} };
				},
			},
			registry: {},
		});
		const sidepanel = renderer.create(
			<TestSidePanel actions={actions} />
		).toJSON();
		expect(sidepanel).toMatchSnapshot();
	});
});
