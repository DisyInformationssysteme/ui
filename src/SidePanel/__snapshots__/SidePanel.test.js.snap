exports[`SidePanel should render provided actions 1`] = `
<div
  className={undefined}>
  <button
    className="btn btn-link"
    disabled={false}
    onClick={[Function bound toggleDock]}
    type="button">
    |
    <i
      className="fa fa-arrow-left" />
  </button>
  <div
    className="btn-group-vertical">
    <a
      className="btn btn-link"
      onClick={[Function bound onClick]}>
      test
    </a>
  </div>
</div>
`;
